From: SeongJae Park <sj@kernel.org>
Date: Thu, 20 Feb 2025 09:35:36 -0800
Subject: [PATCH] ==== add {core,ops}_filters sysfs dirs ====

mm/damon: add sysfs dirs for managing DAMOS filters based on handling layers

DAMOS filters are categorized into two groups based on their handling
layers, namely core and operations layers.  The categorization affects
when each filter is evaluated.  Core layer handled filters are evaluated
first.  The order meant nothing before, but introduction of allow
filters changed that.

DAMOS sysfs interface provides single directory for filters, namely
'filters'.  Users can install any filters in any order there.  DAMON
will internally categorize those into core and operations layer handled
ones, and apply the evaluation order rule.  The ordering rule is clearly
documented.  But the interface could still confuse users since it is
allowed to install filters on the directory in mixed ways.

Add two sysfs directories for managing filters by handling layer, namely
'core_filters' and 'ops_filters' for filters that handled by core and
operations layer, respectively.  Those are avoided to be used for
installing filters that not handled by the assumed layer.

For backward compatibility, keep 'filters' directory and allow mixed
types of filters installation there.  Filters installed in the directory
will be added to DAMON after those of 'core_filters' and 'ops_filters'
directories, with the automatic categorizations.  The 'filters'
directory will start its gradual deprecation once 'core_filters' and
'ops_filters' are proven to be not a bad idea.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_se47dx5u | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_se47dx5u

diff --git a/hkml_cv_bogus/hkml_cv_bogus_se47dx5u b/hkml_cv_bogus/hkml_cv_bogus_se47dx5u
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

